src/OrleansAzureUtils/Hosting/ServiceRuntimeWrapper.cs

```diff
@@ -183,11 +183,9 @@ private void Initialize()
                 logger.Warn(ErrorCode.AzureServiceRuntime_NotLoaded, msg1);
 
                 // Microsoft.WindowsAzure.ServiceRuntime isn't loaded. We may be running within a web role or not in Azure.
-                // Trying to load by partial name, so that we are not version specific.
-                // Assembly.LoadWithPartialName has been deprecated. Is there a better way to load any version of a known assembly?
-                #pragma warning disable 618
-                assembly = Assembly.LoadWithPartialName("Microsoft.WindowsAzure.ServiceRuntime");
-                #pragma warning restore 618
+#pragma warning disable 618
+                assembly = Assembly.Load(new AssemblyName("Microsoft.WindowsAzure.ServiceRuntime, Version = 2.7.0.0, Culture = neutral, PublicKeyToken = 31bf3856ad364e35"));
```

Comment:
We had a conversation about this thing. It turns out that they are not changing the version number of this assembly with the new Azure SDK releases. It was not the case before. Since 2.7 (which is 1 year old) it is still the case, so we decided to remove the magic and just depend on this.